.github/workflows/esp-idf.yml

```diff
@@ -0,0 +1,103 @@
+name: Build ESP-IDF Project
+
+on:
+  push:
+  pull_request:
+
+permissions:
+  contents: read        # Allows reading repository content (needed for checkout)
+  issues: write         # Allows reporting build results to PR checks
+  pull-requests: write  # Allows commenting/updating PR status
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  discover-examples:
+    runs-on: ubuntu-latest
+    outputs:
+      examples: ${{ steps.set-matrix.outputs.examples }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Find examples dynamically
+        id: set-matrix
+        run: |
+          echo "🔍 Searching for top-level example folders..."
+
+          examples=$(find esp_idf/spotflow/examples -mindepth 1 -maxdepth 1 -type d \
+            -exec test -f "{}/CMakeLists.txt" \; -print \
+            | jq -R -c . | jq -s -c .)
+
+          # Ensure we have a JSON array even if no examples found
+          examples=${examples:-[]}
+
+          echo "Examples found: $examples"
+          echo "examples=$examples" >> "$GITHUB_OUTPUT"
+  build:
+    runs-on: ubuntu-latest
+    needs: discover-examples
+    strategy:
+      matrix:
+        target: [esp32s3, esp32c6]
+        example: ${{ fromJson(needs.discover-examples.outputs.examples) }}
+      fail-fast: false
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Verify example path
+        run: |
+          if [ ! -d "${{ matrix.example }}" ]; then
+            echo "❌ Path does not exist: ${{ matrix.example }}"
+            exit 1
+          fi
+          echo "✅ Found example path: ${{ matrix.example }}"
+
+      - name: Cache ESP-IDF installation
+        uses: actions/cache@v3
+        with:
+          path: |
+            $HOME/esp-idf
+            $HOME/.espressif
```

Comment:
Doesn't seem to work - see that it's not present https://github.com/spotflow-io/device-sdk/actions/caches

The latest run shows this warning:
```
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```